src/codeu/chat/client/commandline/Chat.java

```diff
@@ -46,10 +50,18 @@ public Chat(Context context) {
   // the system wants to exit, the function will return false.
   //
   public boolean handleCommand(String line) {
-
-    final Scanner tokens = new Scanner(line.trim());
-
-    final String command = tokens.hasNext() ? tokens.next() : "";
+    final List<String> args = new ArrayList<>();
+    final Tokenizer tokenizer = new Tokenizer(line);
+    
+    try {
+      for (String token = tokenizer.next(); token != null; token = tokenizer.next()) {
+        args.add(token);
+      }
+    } catch (IOException e){ 
+      System.out.println("ERROR! IOException caught in Chat.handleCommand().");
```

Comment:
Generally it's not a good idea to reference file names/line numbers/or method names in an error message like this, because in the future, someone might move the code around or rename the method, and forget to update this line. A better way to communicate this information to the user is to use an exception and a stack trace, which will always be up to date with the files and line numbers.

If you want to make sure this information is included with the error message, try replacing this line with something like:

System.out.println("ERROR! IOException caught.");
e.printStackTrace();